Sources/Networking/HTTPClient/HTTPClient.swift

```diff
@@ -446,9 +466,13 @@ private extension HTTPClient {
                 if httpURLResponse?.isLoadShedder == true {
                     Logger.debug(Strings.network.request_handled_by_load_shedder(request.httpRequest.path))
                 }
+
+                requestRetryScheduled = self.retryRequestIfNeeded(request: request, httpURLResponse: httpURLResponse)
             }
 
-            request.completionHandler?(response)
+            if !requestRetryScheduled {
+                request.completionHandler?(response)
+            }
```

Comment:
Not yet, but this is an excellent thing to add. I'll look into adding the retry attempt number to the headers!

Reply:
Right, we are already tracking http requests response codes: https://github.com/RevenueCat/purchases-ios/blob/373abdd6629ff3eaa3472bd3f442397c11be3f2c/Sources/Diagnostics/DiagnosticsTracker.swift#L26.

We could add the retry count there. I'm a bit reticent to add another tag to the metric in Grafana (we are being mindful of increasing the cardinality of the data), but it might be useful especially if/when we move to using snowflake or similar for this data.
If we want to do this, we just need to append the new property as a valid property for the http_request_performed diagnostic event in the backend and we can start sending it from the SDK. Lmk if you want me to tackle this!